lava-v2-jobs-from-api.py

```diff
@@ -58,17 +70,33 @@ def main(args):
     }
 
     print "Working on kernel %s/%s" % (tree, branch)
-    url = urlparse.urljoin(api, ("/build?job=%s&kernel=%s&git_branch=%s&status=PASS&arch=%s" % (tree, git_describe, branch, arch)))
+    url_params = urllib.urlencode({
+        'job': tree,
+        'kernel': git_describe,
+        'git_branch': branch,
+        'status': 'PASS',
+        'arch': arch,
+    })
+    url = urlparse.urljoin(api, 'build?{}'.format(url_params))
     print "Calling KernelCI API: %s" % url
-    response = requests.get(url, headers=headers)
-    if response.status_code != 200:
-        print "Error calling KernelCI API"
-        print response
-        print response.content
-        sys.exit(1)
-    data = json.loads(response.content)
-    builds = data['result']
+
+    builds = None
+    poll_seconds = 10
+    loops = 600 / poll_seconds # time out after 10 minutes
+    loop = 0
+    while not builds and loop < loops:
```

Comment:
I've discussed this with Milo, I'll drop this polling patch for now and try to fix it on the server side by making the API synchronous.  Doing this should solve the problem, but we need to check it doesn't cause any bad side-effects or any impact on performance.

It's true that polling for only one job is not completely correct either, I guess I didn't really want to expand too much on this work-around.  Hopefully the synchronous API approach will just work.